protoc-gen-swagger/genswagger/template.go

```diff
@@ -1448,6 +1456,104 @@ func processExtensions(inputExts map[string]*structpb.Value) ([]extension, error
 	return exts, nil
 }
 
+func validateHeaderType(headerType string) error {
+	switch headerType {
+	case
+		"string",
+		"number",
+		"integer",
+		"boolean":
+			return nil
+	}
+	return fmt.Errorf("The provided header type: %s. Is not supported", headerType)
+}
+
+func validateDefaultValueType(headerType string, defaultValue string)  (error){
+
+	switch headerType {
+	case
+		"string":
+		if !(isString(defaultValue)){
+			return fmt.Errorf("The provided default value: %s  does not match provider type: %s, or is not properly qouted with escaped quotations", defaultValue, headerType)
+		}
+	case
+		"number":
+			if !(isNumber(defaultValue)){
+				return fmt.Errorf("The provided default value: %s  does not match provider type: %s", defaultValue, headerType)
+			}
+	case
+		"integer":
+			if !(isInt(defaultValue)){
+				return fmt.Errorf("The provided default value: %s  does not match provider type: %s", defaultValue, headerType)
+			}
+	case
+		"boolean":
+			if !(isBool(defaultValue)){
+				return fmt.Errorf("The provided default value: %s  does not match provider type: %s", defaultValue, headerType)
+			}
+	}
+	return nil
+}
+
+func isInt(s string) (bool)  {
+	var digitCheck = regexp.MustCompile(`^[0-9]+$`)
+	return digitCheck.MatchString(s)
+}
+
+func isNumber(s string) (bool)  {
+	var digitCheck = regexp.MustCompile(`^[-+]?[0-9]*\.?[0-9]+([eE][-+]?[0-9]+)?$`)
+	return digitCheck.MatchString(s)
+}
+
+func isString(s string) (bool)  {
+	var digitCheck = regexp.MustCompile(`^"\w+"$`)
+	return digitCheck.MatchString(s)
+}
+
+func isBool(s string) (bool)  {
+	var digitCheck = regexp.MustCompile(`^true|false$`)
+	return digitCheck.MatchString(s)
+}
```

Comment:
https://github.com/OAI/OpenAPI-Specification/blob/3.0.0/versions/2.0.md#headerObject

![image](https://user-images.githubusercontent.com/19742213/95492803-dd546600-0958-11eb-9f2d-eb3e9e1804cf.png)

![image](https://user-images.githubusercontent.com/19742213/95492942-037a0600-0959-11eb-9831-00929dea5b10.png)

Number/Integer: https://swagger.io/docs/specification/data-models/data-types/#numbers 
![image](https://user-images.githubusercontent.com/19742213/95493310-89964c80-0959-11eb-849a-9262616fdf6b.png)
I could probably use `strconv.ParseInt` for the integer if that is preferred. 

Boolean: https://swagger.io/docs/specification/data-models/data-types/#boolean
The reason I used a regex here is because they specifically callout that "truthy" objects are not valid
![image](https://user-images.githubusercontent.com/19742213/95493438-b9ddeb00-0959-11eb-9199-573bca81f393.png)
I believe that strconv.ParseBool returns successfully on truthy values?

String: https://swagger.io/docs/specification/data-models/data-types/#string
This regex was because we were passing in as json.rawMessage objects so all strings would need to follow this format: `^"\w+"$`

Reply:
I converted the number/Integer checks to use strconv. and updated the bool/string checks to use global variables.